tick/optim/prox/src/prox_binarsity.cpp

```diff
@@ -0,0 +1,97 @@
+
+
+#include "prox_tv.h"
+#include "prox_binarsity.h"
+
+
+ProxBinarsity::ProxBinarsity(double strength,
+                             SArrayULongPtr blocks_start,
+                             SArrayULongPtr blocks_length,
+                             bool positive)
+        : Prox(strength, positive), prox_tv_synchronized(false) {
+    // blocks_start and blocks_end have the same size
+    n_blocks = blocks_start->size();
+    if (n_blocks != blocks_length->size()) {
+        throw std::invalid_argument("blocks_start and blocks_length must have the same size");
+    }
+
+    this->blocks_start = blocks_start;
+    this->blocks_length = blocks_length;
+    this->positive = positive;
+    // The object is not ready (synchronize_prox_tv has not been called)
+    prox_tv_synchronized = false;
+}
+
+
+ProxBinarsity::ProxBinarsity(double strength,
+                             SArrayULongPtr blocks_start,
+                             SArrayULongPtr blocks_length,
+                             ulong start,
+                             ulong end, bool positive)
+        : Prox(strength, start, end, positive) {
+    // blocks_start and blocks_end have the same size
+    n_blocks = blocks_start->size();
+    if (n_blocks != blocks_length->size()) {
+        throw std::invalid_argument("blocks_start and blocks_length must have the same size");
+    }
+
+    this->blocks_start = blocks_start;
+    this->blocks_length = blocks_length;
+    this->positive = positive;
+    // The object is not ready (synchronize_prox_tv has not been called)
+    prox_tv_synchronized = false;
+}
+
+
+void ProxBinarsity::synchronize_prox_tv() {
+    if (!prox_tv_synchronized) {
```

Comment:
I would do the opposite:
if `synchronize_prox_tv` is called, you synchronize it without looking at`prox_tv_synchronized`. 
If `value` or `call` is called it looks at `prox_tv_synchronized` to know if it should call `synchronize_prox_tv`